bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java

```diff
@@ -236,23 +236,23 @@ boolean isCorrectlyInitialized() {
      * Request new zone and available input information
      */
     void updateZoneInformation() {
-        updateAsyncMakeOfflineIfFail(zoneAvailableInputs);
-        updateAsyncMakeOfflineIfFail(zoneControl);
+        updateMakeOfflineIfFail(zoneAvailableInputs);
+        updateMakeOfflineIfFail(zoneControl);
 
         if (inputWithPlayControl != null) {
-            updateAsyncMakeOfflineIfFail(inputWithPlayControl);
+            updateMakeOfflineIfFail(inputWithPlayControl);
         }
 
         if (inputWithNavigationControl != null) {
-            updateAsyncMakeOfflineIfFail(inputWithNavigationControl);
+            updateMakeOfflineIfFail(inputWithNavigationControl);
         }
 
         if (inputWithPresetControl != null) {
-            updateAsyncMakeOfflineIfFail(inputWithPresetControl);
+            updateMakeOfflineIfFail(inputWithPresetControl);
         }
 
         if (inputWithDabBandControl != null) {
-            updateAsyncMakeOfflineIfFail(inputWithDabBandControl);
+            updateMakeOfflineIfFail(inputWithDabBandControl);
         }
```

Comment:
Perhaps you could allow the updateMakeOfflineIfFail param to be nullable and just exit early if it is null. That should allow you to get rid of a bunch of these null checks in `updateZoneInformation`.

In addition, you would be able to do something like this:
```
Stream.of(zoneAvailableInputs, zoneControl, inputWithPlayControl, inputWithNavigationControl, inputWithPresetControl, inputWithDabBandControl)
.forEach(this::updateAsyncMakeOfflineIfFail);
```

WDYT?

Reply:
I think it is a good idea.